src/WebJobs.Script/Description/FunctionDescriptor.cs

```diff
@@ -39,6 +41,7 @@ internal FunctionDescriptor()
             TriggerBinding = InputBindings?.SingleOrDefault(p => p.Metadata.IsTrigger);
             HttpTriggerAttribute = GetTriggerAttributeOrNull<HttpTriggerAttribute>();
             LogCategory = LogCategories.CreateFunctionCategory(Name);
+            Logger = loggerFactory.CreateLogger(LogCategory);
```

Comment:
Previously similar to the improvement you're making here, I added LogCategory to cache it. Now that we're caching the logger, LogCategory isn't needed anymore in the public surface it seems. It was only used in that one place in the middleware, which you've removed.

Reply:
ACK - will simplify

Reply:
Also, ILoggerFactory is IDisposable - do we need to make FunctionDescriptor disposable now as well and ensure it's disposed at the right time?

Reply:
Assuming this comes from DI, `ILoggerFactory` is a singleton. So host shutdown handles disposing it.